### PR TITLE
configure openid connect

### PR DESCRIPTION
**Pull request by hahn-kev**
this PR enables the use of openID connect to authenticate 3rd party apps with lexbox. For now this is disabled except in local dev, it's pending key storage decisions. We need to store signing and encryption keys somewhere and I'm not sure what the best option is right now for production.

How to test:
1. open this page https://oidcdebugger.com/
3. enter the following
4. authorize URL: `http://localhost:3000/api/login/open-id-auth`
5. clinet id: `becf2856-0690-434b-b192-a4032b72067f`
6. scope: `openid profile`
7. response type, check only code
8. token URI: http://localhost:3000/api/login/token
9. response mode query
10. click Orange send request button
11. if you're not logged in go to the next step, otherwise go to step 13
12.  you should land at the lexbox login page with a large `ReturnUrl` query parameter in the URL, login like normal (google login unknown?), once login is finished proceed to the next step
13. if you haven't approved this app before go to the next step otherwise go to step 15
14. you should land on a page asking if you want to permit "Oidc Debugger" to access your data: 
image
15. Click Approve to let the debugger have access
16. should arrive at a page like this:
image
17. scroll down and you should see a section labeled ID Token, you should be able to find your name in the payload json object.

if you want to test out the access token (first box, very large token), you can execute a request like this:
```
GET https://localhost:3000/api/AuthTesting/requires-auth
Authorization: Bearer {token here}
```

it should return success with your users name. This represents a token that could be used by a 3rd party app like Language Forge, FLEx lite, or Living Dictionaries to access APIs on behalf of the user.

flow diagram:
```mermaid
flowchart TD
    approve{Has approved already}
    approveUser(Do you approve?)
    start[debugger authorize] -->|Logged in| approve
    start -->|Not logged in| login(Login)
    login --> approve
    approve -->|no| approveUser
    approve -->|yes| debugger
    approveUser -->|"Approve (success)"| debugger
    approveUser -->|"Deny (fail)"| debugger
```

### CODE REVIEW

File: backend/LexBoxApi/Controllers/LoginController.cs [hunk 1]
```diff
@@ -158,6 +172,291 @@ private async Task<ActionResult> EmailLinkExpired()
         return Redirect(returnTo);
     }
 
+    [HttpGet("open-id-auth")]
+    [HttpPost("open-id-auth")]
+    [ProducesResponseType(400)]
+    [ProducesDefaultResponseType]
+    public async Task<ActionResult> Authorize(string? returnUrl = null)
```
[the remark this answers is not in the document]
Comment (hahn-kev) [reply backend/LexBoxApi/Controllers/LoginController.cs:179]: yeah that's probably a good idea. Yeah I copied a fair bit of it from their sample code.

File: frontend/vite.config.ts [hunk 2]
```diff
@@ -30,7 +34,7 @@ export default defineConfig({
     codegen(gqlOptions),
     precompileIntl('src/lib/i18n/locales'),
     sveltekit(),
-    exposeServer ? basicSsl() : null, // crypto.subtle is only availble on secure connections
+    // basicSsl()
```
[the remark this answers is not in the document]
Comment (hahn-kev) [reply frontend/vite.config.ts:37]: originally oauth only worked over ssl so I changed this to always be on, when I figured out how to disable the ssl requirement I disabled the SSL. We could revert this change, though I don't think we ever used the expose server flag did we?

File: backend/LexBoxApi/Auth/AuthKernel.cs [hunk 3]
```diff
@@ -185,6 +196,65 @@ public static class AuthKernel
...
+        services.AddOpenIddict()
+            .AddCore(options =>
+            {
+                options.UseEntityFrameworkCore().UseDbContext<LexBoxDbContext>();
+                options.UseQuartz();
+            })
+            .AddServer(options =>
+            {
+                options.RegisterScopes("openid", "profile", "email");
+                options.RegisterClaims("aud", "email", "exp", "iss", "iat", "sub", "name");
```
[the remark this answers is not in the document]
Comment (hahn-kev) [reply backend/LexBoxApi/Auth/AuthKernel.cs:218]: yes I think so, there might be some extras that we also need.

File: frontend/vite.config.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (myieye) [reply frontend/vite.config.ts:37]: I have used this for debugging and would like to keep it.

**Comment by myieye** [comment]
I did a little design work on this.
Still reviewing...commit coming soon.

image

File: backend/LexBoxApi/Controllers/LoginController.cs (hunk 1 above)
Comment (myieye) [inline backend/LexBoxApi/Controllers/LoginController.cs:179]: Unused
```suggestion
    public async Task<ActionResult> Authorize()
```

**Review by myieye — APPROVED** [review]
That's all the feedback I have. Everything seems to be working.

**Pull request merged**